### PR TITLE
Fix type errors not `TypeError`

### PR DESCRIPTION
**Pull request by jonasdmarkussen**
## Description
Changed errortype from ValueError to TypeError on relevant places in 14 files.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by erlendvollset — CHANGES_REQUESTED** [review]
Invalid Literal _values_ should raise ValueErrors

**Comment by jonasdmarkussen** [comment]
Reverted some of the changes now @erlendvollset - if you want to take another look

**Comment by erlendvollset** [comment]
There are still a bunch of ValueErrors that have been changed to TypeErrors for invalid literal values.

File: cognite/client/_api/data_modeling/instances.py [hunk 1]
```diff
@@ -858,7 +858,7 @@ def histogram(
 
         for histogram in histogram_seq:
             if not isinstance(histogram, Histogram):
-                raise ValueError(f"Not a histogram: {histogram}")
+                raise TypeError(f"Not a histogram: {histogram}")
```
Comment (haakonvt) [inline cognite/client/_api/data_modeling/instances.py:861]: This should be kept (1)

File: cognite/client/_api/raw.py [hunk 2]
```diff
@@ -496,7 +496,7 @@ def _make_columns_param(self, columns: list[str] | None) -> str | None:
         if columns is None:
             return None
         if not isinstance(columns, List):
-            raise ValueError("Expected a list for argument columns")
+            raise TypeError("Expected a list for argument columns")
```
Comment (haakonvt) [inline cognite/client/_api/raw.py:499]: This should be kept (2)

File: cognite/client/_api/sequences.py [hunk 3]
```diff
@@ -925,7 +925,7 @@ def insert(
         elif isinstance(rows, SequenceType) and (len(rows) == 0 or isinstance(rows[0], tuple)):
             all_rows = [{"rowNumber": k, "values": v} for k, v in rows]
         else:
-            raise ValueError("Invalid format for 'rows', expected a list of tuples, list of dict or dict")
+            raise TypeError("Invalid format for 'rows', expected a list of tuples, list of dict or dict")
```
Comment (haakonvt) [inline cognite/client/_api/sequences.py:928]: This should be kept (3)

**Review by haakonvt — COMMENTED** [review]
You have found a lot of wrong exceptions, please re-open this PR 😄 (I commented on the three first, but there are several more!)

**Comment by jonasdmarkussen** [comment]
@haakonvt Updated now - hope I have got it (more) right this time 🥲

File: cognite/client/data_classes/_base.py [hunk 4]
```diff
@@ -580,7 +580,7 @@ def __init__(
         geometries: Collection[Geometry] | None = None,
     ) -> None:
         if type not in self._VALID_TYPES:
-            raise ValueError(f"type must be one of {self._VALID_TYPES}")
+            raise TypeError(f"type must be one of {self._VALID_TYPES}")
```
Comment (haakonvt) [inline cognite/client/data_classes/_base.py:583]: This should stay as type is annotated using `Literal[...]` only 😄 
```suggestion
            raise ValueError(f"type must be one of {self._VALID_TYPES}")
```

File: cognite/client/data_classes/data_modeling/containers.py [hunk 5]
```diff
@@ -218,7 +218,7 @@ class ContainerProperty:
     @classmethod
     def load(cls, data: dict[str, Any]) -> ContainerProperty:
         if "type" not in data:
-            raise ValueError("Type not specified")
+            raise TypeError("Type not specified")
```
Comment (haakonvt) [inline cognite/client/data_classes/data_modeling/containers.py:221]: This is not a type issue - you could argue it should be KeyError though. I think it is best to let it stay ValueError:
```suggestion
            raise ValueError("Type not specified")
```

File: cognite/client/data_classes/data_modeling/containers.py [hunk 6]
```diff
@@ -247,7 +247,7 @@ def load(cls, data: dict) -> RequiresConstraint | UniquenessConstraintDefinition
             return RequiresConstraint.load(data)
         elif data["constraintType"] == "uniqueness":
             return UniquenessConstraintDefinition.load(data)
-        raise ValueError(f"Invalid constraint type {data['constraintType']}")
+        raise TypeError(f"Invalid constraint type {data['constraintType']}")
```
Comment (haakonvt) [inline cognite/client/data_classes/data_modeling/containers.py:250]:
```suggestion
        raise ValueError(f"Invalid constraint type {data['constraintType']}")
```

File: cognite/client/data_classes/data_modeling/containers.py [hunk 7]
```diff
@@ -302,7 +302,7 @@ def load(cls, data: dict) -> Index:
             return BTreeIndex.load(data)
         if data["indexType"] == "inverted":
             return InvertedIndex.load(data)
-        raise ValueError(f"Invalid index type {data['indexType']}")
+        raise TypeError(f"Invalid index type {data['indexType']}")
```
Comment (haakonvt) [inline cognite/client/data_classes/data_modeling/containers.py:305]:
```suggestion
        raise ValueError(f"Invalid index type {data['indexType']}")
```

File: cognite/client/data_classes/data_modeling/data_types.py [hunk 8]
```diff
@@ -79,7 +79,7 @@ def load(cls, data: dict) -> PropertyType:
         elif type_ == "direct":
             return DirectRelation(**data)
 
-        raise ValueError(f"Invalid type {type_}.")
+        raise TypeError(f"Invalid type {type_}.")
```
Comment (haakonvt) [inline cognite/client/data_classes/data_modeling/data_types.py:82]:
```suggestion
        raise ValueError(f"Invalid type {type_}.")
```

File: cognite/client/data_classes/shared.py [hunk 9]
```diff
@@ -81,7 +81,7 @@ def __init__(
         coordinates: list,
     ) -> None:
         if type not in self._VALID_TYPES:
-            raise ValueError(f"type must be one of {self._VALID_TYPES}")
+            raise TypeError(f"type must be one of {self._VALID_TYPES}")
```
Comment (haakonvt) [inline cognite/client/data_classes/shared.py:84]:
```suggestion
            raise ValueError(f"type must be one of {self._VALID_TYPES}")
```

File: cognite/client/data_classes/shared.py [hunk 10]
```diff
@@ -102,7 +102,7 @@ class GeoLocation(dict):
 
     def __init__(self, type: Literal["Feature"], geometry: Geometry, properties: dict | None = None) -> None:
         if type not in self._VALID_TYPES:
-            raise ValueError("Only the 'Feature' type is supported.")
+            raise TypeError("Only the 'Feature' type is supported.")
```
Comment (haakonvt) [inline cognite/client/data_classes/shared.py:105]:
```suggestion
            raise ValueError("Only the 'Feature' type is supported.")
```

File: cognite/client/data_classes/_base.py (hunk 4 above)
Comment (jonasdmarkussen) [reply cognite/client/data_classes/_base.py:583]: Aha, first now I think I get this `Literal` thing - thanks!

**Review by haakonvt — APPROVED** [review]
Lgtm! 🚀

File: cognite/client/data_classes/time_series.py [hunk 11]
```diff
@@ -93,13 +93,13 @@ def count(self) -> int:
             int: The number of datapoints in this time series.
 
         Raises:
-            ValueError: If the time series is string as count aggregate is only supported for numeric data
+            RunTimeError: If the time series is string as count aggregate is only supported for numeric data
```
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:96]:
```suggestion
            RuntimeError: If the time series is string, as count aggregate is only supported for numeric data
```

File: cognite/client/data_classes/time_series.py [hunk 12]
```diff
@@ -93,13 +93,13 @@ def count(self) -> int:
...
 
         Returns:
             int: The total number of datapoints
         """
         if self.is_string:
-            raise ValueError("String time series does not support count aggregate.")
+            raise RunTimeError("String time series does not support count aggregate.")
```
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:102]:
```suggestion
            raise RuntimeError("String time series does not support count aggregate.")
```

**Comment by haakonvt** [comment]
@jonasdmarkussen A few tests that needs to be updated after the changes:
```
FAILED tests/tests_integration/test_api/test_time_series.py::TestTimeSeriesHelperMethods::test_get_count__string_fails - RuntimeError: String time series does not support count aggregate.
FAILED tests/tests_unit/test_api/test_raw.py::TestRawRows::test_list_cols_str_not_supported - TypeError: Expected a list for argument columns
FAILED tests/tests_unit/test_api/test_raw.py::TestRawRows::test_iter_cols_str_not_supported - TypeError: Expected a list for argument columns
FAILED tests/tests_unit/test_data_classes/test_time_series.py::TestTimeSeries::test_get_count__string_raises - RuntimeError: String time series does not support count aggregate.
```

**Review by erlendvollset — APPROVED** [review]
Lgtm 👌👌

**Pull request merged**